### ISSUE TITLE
Rule `no-extra-parens` shouldn't error on logical operators wrapped in parens in a ternary condition

### ISSUE THREAD

**Issue by sindresorhus** [eslint/eslint#6028]
```
~/dev
❯ eslint --version
v2.9.0

~/dev
❯ echo "(foo || bar) ? 'error' : 'warning'" | eslint --stdin --rule='no-extra-parens:2'

<text>
  1:1  error  Gratuitous parentheses around expression  no-extra-parens

✖ 1 problem (1 error, 0 warnings)
```

The ternary `(foo || bar) ? 'error' : 'warning'` without the parens is ambiguous for people not knowing the precedence rules, which applies to a lot of devs. I would have expected the rule not to fail on this.

**Comment by michaelficarra** [comment]
But they are not strictly necessary, so the rule is working properly.

**Comment by domenic** [comment]
I think we need another flag for this. It's a shame "nestedBinaryExpressions" has "binary" in its name, otherwise we could use that.

**Comment by kaicataldo** [comment]
As @michaelficarra said, seems like the rule is working as it should. Are you asking for an option that would relax the rule for the case you outlined above, @sindresorhus? I'm personally not against it, as I think it helps with readability too.

**Comment by eslintbot** [comment]
Hi @sindresorhus, thanks for the issue. It looks like there's not enough information for us to know how to help you. If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by sindresorhus** [comment]
I know it's not strictly necessary, but it improves readability. So I would like an option that allows it.

**Comment by kaicataldo** [comment]
@eslint/eslint-team Other thoughts on this? This hasn't seen much activity, but also didn't get much discussion.

**Comment by nzakas** [comment]
I would like to see an actual proposal, as this is pretty abstract as-is.

**Comment by kaicataldo** [comment]
I'm closing this issues because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.

**Comment by bdougherty** [comment]
How about adding a `"ternaryCondition": false` option that will allow extra parens with a ternary?

```js
/* eslint no-extra-parens: ["error", "all"] */

let something = (foo || bar) ? 'error' : 'warning'; // incorrect
let something = foo || bar ? 'error' : 'warning'; // correct
let somethingElse = (foo === 1) ? 'error' : 'warning'; // incorrect
let somethingElse = foo === 1 ? 'error' : 'warning'; // correct
```

```js
/* eslint no-extra-parens: ["error", "all", { "ternaryCondition": false }] */

let something = (foo || bar) ? 'error' : 'warning'; // correct
let something = foo || bar ? 'error' : 'warning'; // correct
let somethingElse = (foo === 1) ? 'error' : 'warning'; // incorrect
let somethingElse = foo === 1 ? 'error' : 'warning'; // correct
```

**Comment by kaicataldo** [comment]
@bdougherty Would you like to make a new issue with your proposal? You'd have a lot more visibility than you do here!
